Sources/Socket.swift

```diff
@@ -182,6 +182,8 @@ public class Socket {
     public func disconnect(_ callback: (() -> Void)? = nil) {
         connection.delegate = nil
         connection.disconnect()
+
+        onConnectionClosed()
```

Comment:
This will have unintended side effects such as triggering an error and potentially attempting a reconnect. The `onConnectionClosed()` method is built to be called when the connection is disconnected from a call to the `WebSocketDelegate`

I think the best thing to do here is to only perform the required calls
```
self.heartbeatTimer?.invalidate()
self.onCloseCallbacks.forEach( { $0() } )
```

Reply:
Okay, cool, will make those fixes!